internal/config/watchd/watchd_test.go

```diff
@@ -0,0 +1,209 @@
+			var err error
+
+			goldPath := filepath.Join("testdata", "golden", strings.Replace(name, " ", "_", -1))
+			if update {
+				appConfig := watchdconfig.AppConfig{Dirs: tc.wantDirs}
+
+				// Handle error cases
+				if tc.noConfig {
+					// no config file
+				} else if tc.emptyConfig {
+					_, err = os.Create(goldPath)
+					require.NoError(t, err, "failed to create empty config file")
+				} else if tc.noDirs {
+					appConfig.Dirs = nil
+					data, err = yaml.Marshal(&appConfig)
+					require.NoError(t, err, "failed to marshal config")
+					err = os.WriteFile(goldPath, data, 0600)
+					require.NoError(t, err, "failed to write config")
+				} else if tc.badDirs {
+					err = os.WriteFile(goldPath, []byte(`- dirs: "testdir"`), 0600)
+					require.NoError(t, err, "failed to write config")
+				} else {
+					// Normal case
+					data, err = yaml.Marshal(&appConfig)
+					require.NoError(t, err, "failed to marshal config")
+					err = os.WriteFile(goldPath, data, 0600)
+					require.NoError(t, err, "failed to write config")
+				}
+			}
+			got := watchdconfig.GetDirsFromConfigFile(goldPath)
+			require.ElementsMatch(t, tc.wantDirs, got)
+		})
+	}
+}
+
+func TestFilterAbsentDirs(t *testing.T) {
+	tests := map[string]struct {
+		inputDirs    []string
+		existingDirs []string
+
+		wantDirs []string
+	}{
+		"no existing dirs":   {inputDirs: []string{"dir1", "dir2"}},
+		"some existing dirs": {inputDirs: []string{"dir1", "dir2"}, existingDirs: []string{"dir1"}, wantDirs: []string{"dir1"}},
+		"all existing dirs":  {inputDirs: []string{"dir1", "dir2"}, existingDirs: []string{"dir1", "dir2"}, wantDirs: []string{"dir1", "dir2"}},
+	}
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			chdirToTempdir(t)
+			for _, dir := range tc.existingDirs {
+				require.NoError(t, os.MkdirAll(dir, 0750), "failed to create existing dirs")
+			}
+			got := watchdconfig.FilterAbsentDirs(tc.inputDirs)
+			require.ElementsMatch(t, tc.wantDirs, got)
+		})
+	}
+}
+
+func TestWriteConfig(t *testing.T) {
+	tests := map[string]struct {
+		dirs             []string
+		useDefaultConfig bool
+
+		nestedConfigPath bool
+		absentDirs       bool
+
+		wantErr bool
+	}{
+		"with empty dirs":  {wantErr: true},
+		"with absent dirs": {dirs: []string{"dir1", "dir2"}, absentDirs: true, wantErr: true},
+
+		"with relative config path": {dirs: []string{"dir1", "dir2"}},
+		"with nested config path":   {dirs: []string{"dir1", "dir2"}, nestedConfigPath: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			goldPath, err := filepath.Abs(filepath.Join("testdata", "golden", strings.Replace(name, " ", "_", -1)))
+			require.NoError(t, err, "failed to get absolute path")
+
+			chdirToTempdir(t)
+
+			var data []byte
+
+			if !tc.absentDirs {
+				for _, dir := range tc.dirs {
+					require.NoError(t, os.MkdirAll(dir, 0750), "failed to create dirs")
+				}
+			}
+
+			if update {
```

Comment:
Updating the golden files should be just before the comparison at the end, after running the tests and the golden should be the consequence of writing the config here, not preparing manually by another mean.

In some other words, the golden is the result of the code under tests as it’s what you are comparing to.